core/src/test/java/test/retryAnalyzer/dataprovider/RetryAnalyzerWithComplexDataProviderTest.java

```diff
@@ -0,0 +1,55 @@
+package test.retryAnalyzer.dataprovider;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import test.SimpleBaseTest;
+
+public class RetryAnalyzerWithComplexDataProviderTest extends SimpleBaseTest {
+    private static int countWithStringArray = 3;
+    private static int countWithObjectAndStringArray = 3;
+    private static int countWithSingleParam = 3;
+    private static int countWithoutDataProvider = 3;
+
+    @DataProvider(name = "getTestData")
+    public Object[][] getTestData() {
+        return new String[][]{
+                new String[]{"abc1", "cdf1"}};
+    }
+
+    @Test(dataProvider = "getTestData", description = "GITHUB - 2544", retryAnalyzer = DataProviderRetryAnalyzer.class)
+    public void testWithStringArray(String... values) {
+        System.out.println("Test Run " + values);
```

Comment:
Please remove syso

Reply:
Thanks for you comment!!! Pardon me if I am not able to understand your comment properly.

Just wanted to be sure if both of us are on the same page. :)

1. "Here, the tests are expected into many XxxSample classes (one by use-case) and we need to add some tests that will run each use-cases." - Have you meant that we need to add these sample test methods in across classes where scripts were written to test retryAnalyzer functionality(Example - test.retryAnalyzer.dataprovider.issue2163.TestClassPoweredByDataProviderSample.class and test.retryAnalyzer.github1519.TestClassSample.class )? 

**Note -** If this is what you meant just want to understand if this makes any different outcome. Because if all the scripts are executing successfully then this new change does not cause any harm to the existing scripts which were written before.

2. Have you meant to say that the test scripts I have written are not enough to test the dataProvider with retryAnalyzer functionality ?

3. Have you meant that I need to create test class individually for each test-scripts I have written to test ?

Thanks

Reply:
@royanindya-code - What @juherr  is saying is as below:

1. Extract out your test class that is being used as a sample, but which does not extend `SimpleBaseTest`
2. Let that class add its required assertions or use a collection as a point of capturing details.
3. Define a test class that extends `SimpleBaseTest` (this is the class that will find its reference in `src/test/resources/testng.xml` which will create a TestNG object and use its APIs to set the test classes, listener to capture pass/fail information etc., and then it invokes the `TestNG.run()` internally.
4. Define assertions in the test class defined in (3) which validates if TestNG is working fine or not.

@juherr Pls let me know if this captures your expectation

Reply:
@krmahadevan That's a perfect 👍

Reply:
Apologies for my late response. I was little occupied with some pending work. I have updated the PR as mentioned by both of you. Let me know if my understanding is correct. I will close this thread once you approve the change.